### ISSUE TITLE
Options other than Argocd-secret

### ISSUE THREAD

**Issue by yuzujoe** [argoproj/argo-cd#3685]
# Summary
I'm old enough to use Github for dex authentication, but I can only use argocd-secret for secret resource to store clientID and clientSecret value.

It's a very inconvenient specification for those of us who use externally-secret

# Motivation
I want to manage it with the same SECRET.

# Proposal

It seems that you are hardcoding a file that is read by default, but I think you should change it to read a secret resource.
https://github.com/argoproj/argo-cd/search?q=ArgoCDSecretName&unscoped_q=ArgoCDSecretName

**Comment by booleanbetrayal** [comment]
I would also like this functionality for the same reasons.

**Comment by Kampe** [comment]
Yes please!  Updating the same system level argo-cd secrets is *very* bad juju.

**Comment by andersliland** [comment]
Yes please!

**Comment by voidlily** [comment]
I ran into this as well - I use external-secrets to manage github creds, but argocd also wants to store api keys in the same argocd-secret, so external-secrets will overwrite the tokens and lose them, without backing up the tokens to secretsmanager first.

It would be ideal if we could not put argocd-managed things in the same place as externally managed secrets.
